### PR TITLE
WIP: Wrap earthtide

### PR DESCRIPTION
**Pull request by willschlitzer**
This PR wraps the module `earthtide`.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/earthtide.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        - None if ``outgrid`` is set (grid output will be stored in file set by
+          ``outgrid``)
+
+    Example
+    -------
+    >>> import pygmt
+    >>> # Create a grid of Earth tide at 1200 UTC on June 18, 2018
+    >>> grid = pygmt.earthtide(time="2018-06-18T12:00:00")
+    """
+    if "R" in kwargs and "I" not in kwargs:
```
Comment (seisman) [inline pygmt/src/earthtide.py:86]:
```suggestion
    if kwargs.get("R") is not None and kwargs.get("I") is None:
```

File: pygmt/src/earthtide.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        raise GMTInvalidInput("Must specify 'spacing' if 'region' is specified.")
+    with GMTTempFile(suffix=".nc") as tmpfile:
+        with Session() as lib:
+            if "G" not in kwargs:  # if outgrid is unset, output to tempfile
+                kwargs.update({"G": tmpfile.name})
+            outgrid = kwargs["G"]
+            arg_str = build_arg_string(kwargs)
+            lib.call_module("earthtide", arg_str)
```
Comment (seisman) [inline pygmt/src/earthtide.py:94]:
```suggestion
            lib.call_module("earthtide", build_arg_string(kwargs))
```

File: pygmt/src/earthtide.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    >>> # Create a grid of Earth tide at 1200 UTC on June 18, 2018
+    >>> grid = pygmt.earthtide(time="2018-06-18T12:00:00")
+    """
+    if "R" in kwargs and "I" not in kwargs:
+        raise GMTInvalidInput("Must specify 'spacing' if 'region' is specified.")
+    with GMTTempFile(suffix=".nc") as tmpfile:
+        with Session() as lib:
+            if "G" not in kwargs:  # if outgrid is unset, output to tempfile
+                kwargs.update({"G": tmpfile.name})
+            outgrid = kwargs["G"]
```
Comment (seisman) [inline pygmt/src/earthtide.py:92]:
```suggestion
            if (outgrid := kwargs.get("G")) is None:
                kwargs["G"] = outgrid = tmpfile.name  # output to tmpfile
```

File: pygmt/src/earthtide.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    >>> # Create a grid of Earth tide at 1200 UTC on June 18, 2018
+    >>> grid = pygmt.earthtide(time="2018-06-18T12:00:00")
+    """
+    if kwargs.get("R") is not None and kwargs.get("I") is None:
+        raise GMTInvalidInput("Must specify 'spacing' if 'region' is specified.")
+    with GMTTempFile(suffix=".nc") as tmpfile:
+        with Session() as lib:
+            if (outgrid := kwargs.get("G")) is None:
+                kwargs["G"] = outgrid = tmpfile.name  # output to tmpfile
+            lib.call_module("earthtide", build_arg_string(kwargs))
```
Comment (seisman) [inline pygmt/src/earthtide.py:92]:
```suggestion
            lib.call_module(module="earthtide", args=build_arg_string(kwargs))
```

**Comment by willschlitzer** [comment]
I would like to modify this to accept `datetime` objects. Should I use separate variables for specifying a single time and a range of times? I'm not sure how this module is most commonly used, so I don't want to make a hacky/overly complex solution for something that isn't a common need and a text string can work instead.

**Comment by seisman** [comment]
I think it would be much easier to design the wrapper if you can provide some example GMT commands using `earthtide`.

**Comment by willschlitzer** [comment]
> I think it would be much easier to design the wrapper if you can provide some example GMT commands using `earthtide`.

Sorry, what are you asking for? As far as I can tell from the tests, I have a working version of the `earthtide` module that accepts string inputs.

**Comment by willschlitzer** [comment]
Not sure what the issue is with the imports; I'm getting: 
```
module_func.__doc__ = docstring.format(**filler_text)
KeyError: 'I'
```

Since there has been no movement on this for 10 months, I'm going to close the PR.

**Pull request closed without merging**